docs/reference/search/suggesters/context-suggest.asciidoc

```diff
@@ -1,237 +1,269 @@
+    See <<suggester-context-geo>>
 
-However contexts are specified (as type `category` or `geo`, which are discussed below), each
-context value generates a new sub-set of documents which can be queried by the completion
-suggester. All three types accept a `default` parameter which provides a default value to use
-if the corresponding context value is absent.
-
-The basic structure of this element is that each field forms a new context and the fieldname
-is used to reference this context information later on during indexing or querying. All context
-mappings have the `default` and the `type` option in common. The value of the `default` field
-is used, when ever no specific is provided for the certain context. Note that a context is
-defined by at least one value. The `type` option defines the kind of information hold by this
-context. These type will be explained further in the following sections.
+NOTE: Adding context mappings increases the index size for completion field.
 
 [[suggester-context-category]]
 [float]
 ==== Category Context
-The `category` context allows you to specify one or more categories in the document at index time.
-The document will be assigned to each named category, which can then be queried later. The category
-type also allows to specify a field to extract the categories from. The `path` parameter is used to
-specify this field of the documents that should be used. If the referenced field contains multiple
-values, all these values will be used as alternative categories.
+
+The `category` context allows you to associate one or more categories to suggestions at index time.
+At query time, suggestions can be filtered and boosted by their associated categories.
 
 [float]
 ===== Category Mapping
 
-The mapping for a category is simply defined by its `default` values. These can either be
-defined as list of *default* categories:
+A `category` context mapping, where contexts are provided explicitly with suggestions
+can be defined as follows:
 
 [source,js]
 --------------------------------------------------
-"context": {
-    "color": {
+"contexts": [
+    {
+        "name": "cat_context",
         "type": "category",
-        "default": ["red", "orange"]
     }
-}
+]
 --------------------------------------------------
 
-or as a single value
+Alternatively, A `category` context mapping that references another field within a document
+can be defined as follows:
 
 [source,js]
 --------------------------------------------------
-"context": {
-    "color": {
+"contexts": [
+    {
+        "name": "cat_context",
         "type": "category",
-        "default": "red"
+        "path": "cat_field"
     }
-}
+]
 --------------------------------------------------
 
-or as reference to another field within the documents indexed:
+[float]
+===== Indexing category contexts
+
+`category` contexts can be explicitly set with suggestions or be indexed from another field in the
```

Comment:
Category contexts can be specified explicitly when indexing suggestions, or can be retrieved from another 
    field in the document via the `path` parameters. If a suggestion has multiple contexts (categories?),
    the suggestion will be indexed for each category:  (followed by the example)